src/content-script/components/message-list/message-list.tsx

```diff
@@ -25,9 +25,19 @@ type ReduxProps = ConnectedProps<typeof connector>;
 const MessageList: React.FC<ReduxProps> = ({ messages }) => {
   const messageRefs = useRef({} as Record<number, HTMLDivElement | null>);
 
+  const isInViewport = (elem: HTMLDivElement | null) => {
+    const bounds = elem?.getBoundingClientRect();
+    if (!bounds) return true;
+    return (
+      bounds.top >= 0 &&
+      // using a magic number of 130
+      bounds.bottom <= (window.innerHeight || document.documentElement.clientHeight) - 130
```

Comment:
NIT: could you put parenthesis around each conditional statement

Reply:
Yeah no prob, like this?
`(bounds.top >= 0) &&`
`(bounds.bottom <= (window.innerHeight || document.documentElement.clientHeight) - 130)`

Reply:
yup!